mbuild/box.py

```diff
@@ -2,151 +2,333 @@
+
+        return (Lx, Ly, Lz, xy, xz, yz)
+
+    def _get_angles(self):
+        """Calculate the angles between the vectors that define the box.
+
+        Calculates the angles alpha, beta, and gamma from the Box object
+        attribute box_vectors.
+
+        """
+
+        vector_magnitudes = np.linalg.norm(self.box_vectors, axis=1)
+
+        a_dot_b = np.dot(self._box_vectors[0], self._box_vectors[1])
+        b_dot_c = np.dot(self._box_vectors[1], self._box_vectors[2])
+        a_dot_c = np.dot(self._box_vectors[0], self._box_vectors[2])
+
+        alpha_raw = b_dot_c / (vector_magnitudes[1] * vector_magnitudes[2])
+        beta_raw = a_dot_c / (vector_magnitudes[0] * vector_magnitudes[2])
+        gamma_raw = a_dot_b / (vector_magnitudes[0] * vector_magnitudes[1])
+
+        alpha = np.rad2deg(np.arccos(np.clip(alpha_raw, -1.0, 1.0)))
+        beta = np.rad2deg(np.arccos(np.clip(beta_raw, -1.0, 1.0)))
+        gamma = np.rad2deg(np.arccos(np.clip(gamma_raw, -1.0, 1.0)))
+
+        return (alpha, beta, gamma)
+
+def _validate_box_vectors(box_vectors):
+    """Determine if the vectors are in the convention we use.
+
+    This method will parse the inputted box vectors, determine if the
+    vectors follow the conventions the Box class adheres to. In this case:
+
+    1. It is a 3x3 matrix that can be coerced into a numpy array of floats
+    2. Vectors are in a right-handed basis (determinant of matrix is +)
+    3. The first vector is aligned along the [1,0,0] direction
+    4. The second vector in aligned along the xy plane
+    5. The third vector can align freely in the x,y, and +z direction
+
+    If the vectors are not right-handed, a warning will be raised,
+    and the vectors will be transformed into the right-handed coordinate
+    system.
+
+    If the three vectors are not following conventions 3-5, the matrix will
+    be transformed to comply with them, and will also raise a warning.
     """
-    def __new__(cls, array, var=None, box=None, dtype=np.float):
-        _array = np.asarray(array, dtype).view(cls)
-        _array.var = var
-        _array.box = box
-        return _array
-
-    def __setitem__(self, key, val):
-        array = list(self)
-        array[key] = val
-        if self.var == "maxs":
-            msg = "Given max value is less than box's min value" 
-            assert val >= self.box.mins[key], msg
-            self.box.maxs = array
-        elif self.var == "mins":
-            msg = "Given min value is more than box's max value"
-            assert val <= self.box.maxs[key], msg
-            self.box.mins = array
-        elif self.var == "lengths":
-            msg = "Given length value is negative"
-            assert val >= 0, msg
-            self.box.lengths = array
-        else:
-            self.box.angles = array
+    vecs = np.asarray(box_vectors, dtype=np.float)
+    if vecs.shape == (3,3):
+        pass
+    else:
+        vecs.reshape(3,3)
```

Comment:
It's fine to call this reshape regardless of the original shape.
```suggestion
    vecs.reshape(3,3)
```